perceval/runtime/job_group.py

```diff
@@ -0,0 +1,281 @@
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import warnings
+
+from perceval.runtime import RemoteJob
+from perceval.runtime.rpc_handler import RPCHandler
+from perceval.utils import PersistentData, FileFormat
+from perceval.utils.logging import get_logger, channel
+import os
+import json
+import datetime
+
+FILE_EXT_JGRP = 'jgrp'
+MAX_SAMPLE_DEF_GROUP = 10000
+
+class JobGroup:
+    """
+    A JobGroup handles a collection of Jobs saved on disk using perceval
+    persistent data. It can perform various tasks such as
+    - Saving information for a collection of jobs, whether they have been sent to the cloud or not.
+    - Running jobs within the group either in parallel or sequentially.
+    - Rerunning failed jobs within the group.
+    """
+    def __init__(self, name: str):
+        self._name = name
+        self._group_info = dict()
+        self._file_name = f"{self._name}.{FILE_EXT_JGRP}"
+
+        if self._job_group_exists(name):
+            get_logger().info(f'Job Group with name {name} exists, loading from disk', channel.general)
+            self._load_job_group()
+        else:
+            self._create_job_group()
+
+
+    @property
+    def name(self):
+        """
+        Name of the job Group [also the filename (without extension) on disk]
+        """
+        return self._name
+
+    @property
+    def job_group_data(self):
+        """
+        List of necessary information (id,status,metadata) for each job within the group
+        """
+        return self._group_info['job_group_data']
+
+    @staticmethod
+    def _get_current_datetime():
+        return datetime.datetime.now().strftime("%Y%m%d_%H%M%S")
+
+    @staticmethod
+    def _get_job_group_dir():
+        return os.path.join(PersistentData().directory, "job_group")
+
+    @staticmethod
+    def _setup_job_group_dir(dir_path):
+        # creates a sub folder in persistent data directory if non-existent
+        try:
```

Comment:
This code should probably be added to `PersistentData`. I'd suggest a `create_sub_directory(relative_path)` method that you'd call with a path relative to the root of persistent data.
In your case:  `self._persistent_data.create_sub_directory("job_group")`